pkg/daemon/update.go

```diff
@@ -975,11 +1007,21 @@ func (dn *Daemon) disableUnit(unit igntypes.Unit) error {
 
 // writeUnits writes the systemd units to disk
 func (dn *Daemon) writeUnits(units []igntypes.Unit) error {
+	operatingSystem, err := getHostRunningOS()
+	if err != nil {
+		return errors.Wrapf(err, "checking operating system")
+	}
 	for _, u := range units {
 		// write the dropin to disk
 		for i := range u.Dropins {
 			glog.Infof("Writing systemd unit dropin %q", u.Dropins[i].Name)
 			dpath := filepath.Join(pathSystemd, u.Name+".d", u.Dropins[i].Name)
+			if _, err := os.Stat("/usr" + dpath); err == nil &&
+				(operatingSystem == machineConfigDaemonOSRHCOS || operatingSystem == machineConfigDaemonOSFCOS) {
+				if err := createOrigFile(dpath); err != nil {
```

Comment:
So by doing this instead we're running into another erroneous (although no longer regressive?) behaviour:

If we look at the workflow I commented above (Create MC1, upgrade to this PR, create MC2), but with a slight difference: there originally exists a version of the service on disk.

So now when we get here, we realize that there was an original version and attempt to create a backup. Instead of backing up from what existed on disk at the beginning, we backup what MC1 dropped since we're copying from `$dpath` directly.

I think we need to modify behaviour so that we create the orig file from the `/usr/$dpath` location instead.

Reply:
We would probably need to modify the `createOrigFile` function to take a "from" location and a "to" location

Reply:
so in this case, we always want the original from `/usr/$dpath` right? should I just `createOrigFile("/usr"+$dpath)`?

Reply:
```diff
diff --git a/pkg/daemon/update.go b/pkg/daemon/update.go
index 492e0f79..cadfc7d1 100644
--- a/pkg/daemon/update.go
+++ b/pkg/daemon/update.go
@@ -1018,7 +1018,7 @@ func (dn *Daemon) writeUnits(units []igntypes.Unit) error {
                        dpath := filepath.Join(pathSystemd, u.Name+".d", u.Dropins[i].Name)
                        if _, err := os.Stat("/usr" + dpath); err == nil &&
                                (operatingSystem == machineConfigDaemonOSRHCOS || operatingSystem == machineConfigDaemonOSFCOS) {
-                               if err := createOrigFile(dpath); err != nil {
+                               if err := createOrigFile("/usr"+dpath); err != nil {
                                        return err
                                }
                        }
@@ -1052,7 +1052,7 @@ func (dn *Daemon) writeUnits(units []igntypes.Unit) error {
                        glog.Infof("Writing systemd unit %q", u.Name)
                        if _, err := os.Stat("/usr" + fpath); err == nil &&
                                (operatingSystem == machineConfigDaemonOSRHCOS || operatingSystem == machineConfigDaemonOSFCOS) {
-                               if err := createOrigFile(fpath); err != nil {
+                               if err := createOrigFile("/usr" + fpath); err != nil {
                                        return err
                                }
                        }
```

Reply:
Wouldn't it back up `/usr/$dpath` so it writes back to `/usr/$dpath` when it comes time to restore? Then we don't restore `$dpath`

Basically we currently have a forced 1-to-1 between the location of file backed up and the origin location of it. What we want is to back up `/usr/$dpath` to `$dpath` when we restore, and not restore `/usr/$dpath` to `/usr/$dpath`

Reply:
alrighty, I think I've modified createOrigFile to take a fromPath arg and backup _from_ there once we're backing up dropins and unit